taxonomy-manager-rest-server/src/main/java/com/digirati/taxman/rest/server/taxonomy/ConceptModelRepository.java

```diff
@@ -64,6 +68,7 @@ public ConceptModel create(ConceptModel model) {
 
         var dataset = dataMapper.map(model);
         conceptDao.storeDataSet(dataset);
+        conceptExtractionService.getTextLookupService().addConcept(model);
```

Comment:
The encapsulating transaction can't complete until the automaton is rebuilt in this case. You need to dispatch an event and handle the concept being added asynchronously instead.

Something like:
```java
@Inject @Stream(“createdConceptModels”)
Emitter<ConceptModel> createdEmitter;

createdEmitter.send(model);
```

See reactive messaging: https://lordofthejars.github.io/quarkus-cheat-sheet/#_version_0_15_0

Reply:
Unfortunately it looks like there was a regression in Quarkus v0.16.0 which stops the application from even starting when you add the required extension (smallrye-reactive-messaging). It appears to be a race condition between the class loader and one of the logger factories but I've not gotten to the bottom of it yet. I've downgraded locally to 0.15.0 and the application starts correctly but for some reason doesn't inject the repository into the resource. So I'm going to park this until I've resolved the other comments then I'll try and track down exactly what's going wrong.

Reply:
Here's the stacktrace:

```
14:40:40,286 ERROR [io.qua.dev.DevModeMain] Failed to start quarkus: java.lang.RuntimeException: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
        [error]: Build step io.quarkus.netty.deployment.NettyProcessor#build threw an exception: java.lang.ExceptionInInitializerError
        at io.quarkus.runner.RuntimeRunner.run(RuntimeRunner.java:132)
        at io.quarkus.dev.DevModeMain.doStart(DevModeMain.java:166)
        at io.quarkus.dev.DevModeMain.main(DevModeMain.java:88)
Caused by: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
        [error]: Build step io.quarkus.netty.deployment.NettyProcessor#build threw an exception: java.lang.ExceptionInInitializerError
        at io.quarkus.builder.Execution.run(Execution.java:108)
        at io.quarkus.builder.BuildExecutionBuilder.execute(BuildExecutionBuilder.java:121)
        at io.quarkus.deployment.QuarkusAugmentor.run(QuarkusAugmentor.java:104)
        at io.quarkus.runner.RuntimeRunner.run(RuntimeRunner.java:98)
        ... 2 more
Caused by: java.lang.ExceptionInInitializerError
        at io.netty.buffer.AbstractByteBufAllocator.<clinit>(AbstractByteBufAllocator.java:36)
        at io.netty.buffer.Unpooled.<clinit>(Unpooled.java:73)
        at io.netty.handler.codec.http.HttpObjectEncoder.<clinit>(HttpObjectEncoder.java:53)
        at java.base/java.lang.Class.forName0(Native Method)
        at java.base/java.lang.Class.forName(Class.java:315)
        at io.quarkus.netty.deployment.NettyProcessor.build(NettyProcessor.java:42)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:566)
        at io.quarkus.deployment.ExtensionLoader$1.execute(ExtensionLoader.java:507)
        at io.quarkus.builder.BuildContext.run(BuildContext.java:398)
        at org.jboss.threads.ContextClassLoaderSavingRunnable.run(ContextClassLoaderSavingRunnable.java:35)
        at org.jboss.threads.EnhancedQueueExecutor.safeRun(EnhancedQueueExecutor.java:2011)
        at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.doRunTask(EnhancedQueueExecutor.java:1538)
        at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1395)
        at java.base/java.lang.Thread.run(Thread.java:834)
        at org.jboss.threads.JBossThread.run(JBossThread.java:479)
Caused by: java.lang.NullPointerException: name
        at io.netty.util.internal.logging.AbstractInternalLogger.<init>(AbstractInternalLogger.java:41)
        at io.netty.util.internal.logging.LocationAwareSlf4JLogger.<init>(LocationAwareSlf4JLogger.java:37)
        at io.netty.util.internal.logging.Slf4JLoggerFactory.wrapLogger(Slf4JLoggerFactory.java:54)
        at io.netty.util.internal.logging.Slf4JLoggerFactory.newInstance(Slf4JLoggerFactory.java:49)
        at io.netty.util.internal.logging.InternalLoggerFactory.getInstance(InternalLoggerFactory.java:93)
        at io.netty.util.internal.logging.InternalLoggerFactory.getInstance(InternalLoggerFactory.java:86)
        at io.netty.util.ResourceLeakDetector.<clinit>(ResourceLeakDetector.java:101)
        ... 18 more

```